### PR TITLE
fix: version comparison function based on  PEP440

### LINKED ISSUE #2135: Bug: isManagerVersionCompatibleWith
The isManagerVersionCompatibleWith function, which compares the manager version, doesn't work properly when the version has non-numeric strings such as 'dev' and 'rc'.

It should follow the [PEP 440](https://packaging.python.org/en/latest/specifications/version-specifiers/#summary-of-permitted-suffixes-and-relative-ordering).

ref: [thread 1](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1702870643766?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1702867182940&teamName=devops&channelName=Frontend&createdTime=1702870643766),[ thread 2](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1704698232050?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1704697680477&teamName=devops&channelName=Frontend&createdTime=1704698232050)

### PR DESCRIPTION
**Pull request by yomybaby**
resolve #2135 

This PR introduces
- `comparePEP440Versions`  ([ref](https://packaging.python.org/en/latest/specifications/version-specifiers/#summary-of-permitted-suffixes-and-relative-ordering))
- `backend.ai-client-esm.ts` uses `comparePEP440Versions`
- tests for `comparePEP440Versions`
- Jest configuration for test files in /src (Lit part)
  - You can run tests for the Lit part by running `npm run jest` in the root directory.

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

---

copilot:summary

### CODE REVIEW

File: src/lib/pep440.test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+describe('comparePEP440Versions', () => {
+  test('should compare versions correctly', () => {
+    expect(comparePEP440Versions('1.2.3', '1.2.3')).toBe(0);
+    expect(comparePEP440Versions('1.2.3', '1.2.4')).toBe(-1);
+    expect(comparePEP440Versions('1.2.4', '1.2.3')).toBe(1);
+    expect(comparePEP440Versions('1.2.3.dev.1', '1.2.3')).toBe(-1);
+    expect(comparePEP440Versions('1.2.3', '1.2.3.dev.1')).toBe(1);
+    expect(comparePEP440Versions('1.2.3.dev.1', '1.2.3.dev.2')).toBe(-1);
+    expect(comparePEP440Versions('1.2.3.dev.2', '1.2.3.dev.1')).toBe(1);
+    expect(comparePEP440Versions('1.2.3.dev.2', '1.2.3.dev.1')).toBe(1);
```
Comment (agatha197) [inline src/lib/pep440.test.ts]: It's the same code as line 22.

**Review by agatha197 — CHANGES_REQUESTED** [review]
I just find same code (22, 23 line). And except that, LGTM!

File: src/lib/pep440.test.ts (hunk 1 above)
Comment (agatha197) [inline src/lib/pep440.test.ts]:
```suggestion
```

**Review by agatha197 — CHANGES_REQUESTED** [review]

**Comment by yomybaby** [comment]
@agatha197  Thank you for reviewing. I just pushed a commit you suggested. 
FYI, I'll merge this after releasing a new version (23.09.8)

**Review by agatha197 — APPROVED** [review]
LGTM

**Comment by yomybaby** [comment]
a gude for this PR ([Teams](https://teams.microsoft.com/l/message/19:14c484402d874dafb15806d093b95a82@thread.skype/1706602112700?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1706602112700&teamName=devops&channelName=Frontend&createdTime=1706602112700))

**Pull request merged**